src/outagesim/backup_reliability.jl

```diff
@@ -505,126 +768,325 @@ function backup_reliability_inputs(d::Dict, p::REoptInputs; r::Dict = Dict())::D
+            if !haskey(r2, :battery_size_kwh)
+                push!(invalid_args, "Battery kW provided to reliability inputs but no kWh provided.")
+            end
+        end
+    end
+
+    if length(invalid_args) > 0
+        error("Invalid argument values: $(invalid_args)")
+    end
+
+    return r2
+end
+"""
+    return_backup_reliability(; critical_loads_kw::Vector, generator_operational_availability::Real = 0.9998, generator_failure_to_start::Real = 0.0066, 
+        generator_failure_to_run::Real = 0.00157, num_generators::Int = 1, generator_size_kw::Real = 0.0, num_battery_bins::Int = 100, max_outage_duration::Int = 96,
+        battery_size_kw::Real = 0.0, battery_size_kwh::Real = 0.0, battery_charge_efficiency::Real = 0.948, battery_discharge_efficiency::Real = 0.948)::Array
+
+Return an array of backup reliability calculations. Inputs can be unpacked from backup_reliability_inputs() dictionary
+# Arguments
+-net_critical_loads_kw::Vector                                                      vector of net critical loads                     
+-generator_operational_availability::Union{Real, Vector{<:Real}}      = 0.9998        Fraction of year generators not down for maintenance
+-generator_failure_to_start::Union{Real, Vector{<:Real}}                        = 0.0066        Chance of generator starting given outage
+-generator_failure_to_run::Union{Real, Vector{<:Real}}                = 0.00157       Chance of generator failing in each hour of outage
+-num_generators::Union{Int, Vector{Int}}                            = 1             Number of generators
+-generator_size_kw::Union{Real, Vector{<:Real}}                       = 0.0           Backup generator capacity
+-num_battery_bins::Int              = 100          Internal value for modeling battery
+-max_outage_duration::Int           = 96           Maximum outage hour modeled
+-battery_size_kw::Real              = 0.0          Battery kW of power capacity
+-battery_size_kwh::Real             = 0.0          Battery kWh of energy capacity
+-battery_charge_efficiency::Real    = 0.948        Efficiency of charging battery
+-battery_discharge_efficiency::Real = 0.948        Efficiency of discharging battery
+```
+"""
+function return_backup_reliability(; 
+    net_critical_loads_kw::Vector, 
+    battery_starting_soc_kwh::Array = [],  
+    generator_operational_availability::Union{Real, Vector{<:Real}} = 0.9998, 
+    generator_failure_to_start::Union{Real, Vector{<:Real}}  = 0.0066, 
+    generator_failure_to_run::Union{Real, Vector{<:Real}}  = 0.00157, 
+    num_generators::Union{Int, Vector{Int}} = 1, 
+    generator_size_kw::Union{Real, Vector{<:Real}} = 0.0, 
+    num_battery_bins::Int = 101,
+    max_outage_duration::Int = 96,
+    battery_size_kw::Real = 0.0,
+    battery_size_kwh::Real = 0.0,
+    battery_charge_efficiency::Real = 0.948, 
+    battery_discharge_efficiency::Real = 0.948,
+    kwargs...)::Array
+ 
+    total_gen_cap = sum(generator_size_kw)
```

Comment:
unused variable

Reply:
Was an artifact of a previous version. Just pushed an update which removes it